### PR TITLE
feat(Array): add support for join

### PR DESCRIPTION
**Pull request by vicb**
@mprobst could you please review

### CODE REVIEW

File: lib/facade_converter.ts [hunk 1]
```diff
@@ -303,6 +303,14 @@ export class FacadeConverter extends base.TranspilerBase {
       });
       this.emit(')');
     },
+    'Array.join': (c: ts.CallExpression, context: ts.Expression) => {
+      this.visit(context);
+      if (c.arguments.length === 0) {
```
Comment (mprobst) [inline lib/facade_converter.ts]: nit: I'm ok with writing `if (c.arguments.length) { ... }`, that's idiomatic enough in my opinion. YMMV.

File: test/facade_converter_test.ts [hunk 2]
```diff
@@ -94,32 +94,46 @@ describe('type based translation', () => {
     it('translates array operations to dartisms', () => {
       expectWithTypes('var x: Array<number> = []; x.push(1); x.pop();')
           .to.equal(' List < num > x = [ ] ; x . add ( 1 ) ; x . removeLast ( ) ;');
+
```
Comment (mprobst) [inline test/facade_converter_test.ts]: Why the newlines here, but not below? I think I'd rather not have many newlines, but group related blocks of tests within this `it` by a newline.

**Comment by mprobst** [comment]
LGTM.

File: lib/facade_converter.ts (hunk 1 above)
Comment (vicb) [inline lib/facade_converter.ts]: right, we don't need to transpile the transpiler, I'll update

File: test/facade_converter_test.ts (hunk 2 above)
Comment (vicb) [inline test/facade_converter_test.ts]: I'll remove the newlines from here (indeed, the idea was to group related tests which explain why there is no newlines between `unshift` and `unshift`)

**Comment by vicb** [comment]
landed as b642cf10d01afed7f7b67e93c0d936cdf6fdcc84

**Pull request closed without merging**